test/unit/mc/test_cholesky.cpp

```diff
@@ -197,3 +209,37 @@ TYPED_TEST(CholeskyDistributedTest, NoSquareBlockException) {
     }
   }
 }
+
+TYPED_TEST(CholeskyDistributedTest, MatrixNotDistributedOnGridException) {
+  // Test: matrix not distributed according to the given communicator grid
+  for (const auto& comm_grid : this->commGrids()) {
+    for (const auto& size : square_sizes) {
+      for (const auto& block_size : square_block_sizes) {
+        GlobalElementSize sz = globalTestSize(size);
+
+        {
+          // Different grid size
+          Size2D grid_distributed = Size2D(comm_grid.size().rows() + 1, comm_grid.size().cols() + 1);
+          Distribution distribution(sz, block_size, grid_distributed, comm_grid.rank(), {0, 0});
+          Matrix<TypeParam, Device::CPU> mat(std::move(distribution));
+
+          EXPECT_THROW(cholesky(comm_grid, blas::Uplo::Lower, mat), std::invalid_argument);
+        }
+
+        {
+          // Different rank
+          dlaf::comm::Index2D rank_index_distributed(std::max(0, comm_grid.rank().row() - 1),
```

Comment:
Size2D and Index2D are both in namespace `dlaf::comm::`.
Please be consistent in namespace usage (see general comment).